Scala3 changes

viperey opened the pull request:
Submitting so far some initial work on a scala3 version.
I'm not certain if you are still around and you are interested in this change.
If that's the case, I can work further on polishing the PR.

Let me know.
Best.

bartholomews commented on README.md:
```diff
@@ -1,6 +1,11 @@
 [![CircleCI](https://circleci.com/gh/bartholomews/scalatestudo/tree/master.svg?style=svg)](https://circleci.com/gh/bartholomews/scala-iso/tree/master)
 [![License: MIT](https://img.shields.io/badge/License-MIT-blue.svg)](https://github.com/bartholomews/spotify4s/blob/master/LICENSE)
 
+# Disclaimer
+
+Dirty fork from: https://github.com/bartholomews/scala-iso
+Sole purpose is to gain a scala3 release since the upstream project seems abandoned.
+
```
I guess this is not relevant if it's going back here ?

viperey replied:
:+1:

bartholomews commented on build.sbt:
```diff
@@ -1,24 +1,27 @@
 name := "scala-iso"
-scalaVersion := "2.13.3"
-crossScalaVersions := Seq("2.12.13", "2.13.3")
```
note - I still would like to cross release (to 2.13 mostly, don't mind about 2.12 tbh) - I think it should work having both  3 and 2_13 in `crossScalaVersions`

viperey replied:
:+1:

bartholomews commented on build.sbt:
```diff
@@ -1,24 +1,27 @@
-))
+)
+
+scalacOptions += "-Yretain-trees"
```
to make it work in cross version (i.e. `+compile`) this should work
```
scalacOptions ++= {
  CrossVersion.partialVersion(scalaVersion.value) match {
    case Some((3, _)) => Seq("-Yretain-trees")
    case Some((2, _)) => Seq.empty
    case _ => Seq.empty
  }
}
```

viperey replied:
:+1:

bartholomews reviewed (commented):
thanks for the PR, much appreciated 👍  - I forgot about this repo but happy to have this in and release for scala 3 (I haven't really used yet tbh) - but I would like to cross compile to 2_13 as well

(ps - I will need to fix the CI because `sbt-ci-release` / sonatype is out of date but can do it as a follow-up to this)

viperey commented:
Addressed the comments, all logical (made the initial submission with dirty work since I wasn't sure you were still active)
Checked locally with:
```
sbt ++2.13.16 clean publishLocal
sbt ++3.7.0 clean publishLocal
```
Both work.

Hope this is polished enough on my end for you to easily release the version.
Please ping me back when the release takes place, so I can use the official release rather the locally compiled one.
Thanks!

bartholomews reviewed (approved):
LGTM

bartholomews commented:
@viperey FYI I have released `0.2.0` https://central.sonatype.com/artifact/io.bartholomews/scala-iso_3

viperey commented:
Great, thanks for letting me know :)

Since we are at it, apologies for the off-topic, would you be interested in a PR that also adds ISO 639-3 language codes?

bartholomews commented:
> would you be interested in a PR that also adds ISO 639-3 language codes?

sounds good 👍  thanks
